swarm/network/protocol_test.go

```diff
@@ -225,16 +240,59 @@ func TestBzzHandshakeVersionMismatch(t *testing.T) {
 }
 
 func TestBzzHandshakeSuccess(t *testing.T) {
```

Comment:
The following 3 tests are basically identical. I think this is introducing too much code with too little test value. Please create one test with a table-driven params, when you have both options - light node on/off. https://github.com/golang/go/wiki/TableDrivenTests